openhtf/exe/test_state.py

```diff
@@ -72,17 +72,20 @@ def __init__(self, config, test, plugs, dut_id):
             'filename': test.filename,
             'docstring': test.docstring
             })
-    self.logger = logging.getLogger(logs.RECORD_LOGGER)
-    self.logger.propagate = False
-    self.logger.setLevel(logging.DEBUG)  # Let the handler do the filtering.
-    self.logger.addHandler(logs.RecordHandler(self.record))
-    self.logger.addHandler(logging.StreamHandler(stream=sys.stdout))
-    self.phase_data = phase_data.PhaseData(self.logger, config, plugs,
+    self._record_handler = logs.RecordHandler(self.record)
+    self._stream_handler = logging.StreamHandler(stream=sys.stdout)
```

Comment:
sorry, i think maybe i was confusing in my last comment, here should look like this:
```
...
self.logger = logging.getLogger(logs.RECORD_LOGGER)
self._record_handler = logs.RecordHandler(self.record)
self.logger.addHandler(self._record_handler)
self.phase_data = phase_data.PhaseData(self.logger, config, plugs, self.record)
...
```
And then in `__del__()` and `GetFinishedRecord()` just `self.logger.removeHandler(self._record_handler)` (leave the stream handler).

And in `setup_logger()` around https://github.com/google/openhtf/blob/master/openhtf/util/logs.py#L137 you'll want to do:
```
record_logger = logging.getLogger(RECORD_LOGGER)
record_logger.propagate = False
record_logger.setLevel(logging.DEBUG)
record_logger.addHandler(logging.StreamHandler(stream=sys.stdout))
```
Probably want to do that at the end of setup_logger(), after the root 'openhtf' logger has been set up, but it doesn't really matter.